hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java

```diff
@@ -36,47 +48,77 @@
  */
 @Singleton
 public class NodeUpdateHandler implements TransactionHandler {
+    private final AddressBookValidator addressBookValidator;
 
     @Inject
-    public NodeUpdateHandler() {
-        // Exists for injection
+    public NodeUpdateHandler(@NonNull final AddressBookValidator addressBookValidator) {
+        this.addressBookValidator =
+                requireNonNull(addressBookValidator, "The supplied argument 'addressBookValidator' must not be null");
     }
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
         final var op = txn.nodeUpdate();
-        throw new UnsupportedOperationException("need implementation");
+        validateFalsePreCheck(op.nodeId() < 0, INVALID_NODE_ID);
+        if (op.hasAccountId()) {
+            final var accountId = validateAccountID(op.accountIdOrElse(AccountID.DEFAULT), INVALID_NODE_ACCOUNT_ID);
+            validateFalsePreCheck(!accountId.hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
+        }
+        if (op.hasGossipCaCertificate())
+            validateFalsePreCheck(op.gossipCaCertificate().equals(Bytes.EMPTY), INVALID_GOSSIP_CAE_CERTIFICATE);
     }
 
     @Override
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
-        final var op = context.body().nodeUpdate();
-        throw new UnsupportedOperationException("need implementation");
     }
 
-    /**
-     * Given the appropriate context, updates a topic.
-     *
-     * @param handleContext the {@link HandleContext} for the active transaction
-     * @throws NullPointerException if one of the arguments is {@code null}
-     */
     @Override
     public void handle(@NonNull final HandleContext handleContext) {
         requireNonNull(handleContext);
+        final var op = handleContext.body().nodeUpdate();
 
-        final var txn = handleContext.body();
-        final var op = txn.nodeUpdate();
-
+        final var configuration = handleContext.configuration();
+        final var nodeConfig = configuration.getConfigData(NodesConfig.class);
         final var nodeStore = handleContext.writableStore(WritableNodeStore.class);
-        throw new UnsupportedOperationException("need implementation");
+        final var accountStore = handleContext.readableStore(ReadableAccountStore.class);
+
+        final var existingNode = nodeStore.get(op.nodeId());
+        validateFalse(existingNode == null, INVALID_NODE_ID);
+        if (op.hasAccountId()) {
+            final var accountId = op.accountIdOrElse(AccountID.DEFAULT);
```

Comment:
When using `hasAccountId` we should use `OrThrow` here

Reply:
if (op.hasAccountId()), means accountId != null, accountIdOrThrow() and op.accountIdOrElse(AccountID.DEFAULT) will return the same value. I will change it in the next PR since it's usual way.